File: leetcode/2849/solution.go
/**
 * https://leetcode.com/problems/determine-if-a-cell-is-reachable-at-a-given-time/
 * 2849. Determine if a Cell is Reachable at a Given Time
 */
package main

func isReachableAtTime(sx int, sy int, fx int, fy int, t int) bool {
	steps := max(abs(sx, fx), abs(sy, fy))
	if t >= 2 {
		return steps <= t
	}
	if t == 0 {
		return steps == 0
	}
	// t == 1
	return steps > 0 && steps <= t
}

func abs(a, b int) int {
	if a > b {
		return a - b
	}
	return b - a
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: leetcode/2849/solution.rs
/**
 * https://leetcode.com/problems/determine-if-a-cell-is-reachable-at-a-given-time/
 * 2849. Determine if a Cell is Reachable at a Given Time
 */

impl Solution {
    pub fn is_reachable_at_time(sx: i32, sy: i32, fx: i32, fy: i32, t: i32) -> bool {
        let steps = (sx - fx).abs().max((sy - fy).abs());
        match t {
            0 => steps == 0,
            1 => steps > 0 && steps <= t,
            _ => steps <= t,
        }
    }
}
